randovania/games/planets_zebeth/exporter/patch_data_factory.py

```diff
@@ -90,43 +104,88 @@ def _create_cosmetics(self) -> dict:
             "room_names_on_hud": c.show_room_names.value,
         }
 
-    def _get_item_data(self):
-        item_data = json_lib.read_path(
+    def _get_item_data(self) -> dict:
+        item_data: dict = json_lib.read_path(
             RandovaniaGame.METROID_PLANETS_ZEBETH.data_path.joinpath("pickup_database", "item_data.json")
         )
 
+        for i in range(1, 10):
+            item_data[f"Tourian Key {i}"] = item_data["Tourian Key"]
         item_data["Missiles"] = item_data["Missile Tank"]
         item_data["EnergyTransferModule"] = item_data["Nothing"]
         return item_data
 
-    def _create_object_datas(self):
-        pass
-
     def game_enum(self) -> RandovaniaGame:
         return RandovaniaGame.METROID_PLANETS_ZEBETH
 
-    def create_data(self) -> dict:
-        db = self.game
-
-        useless_target = PickupTarget(
-            pickup_creator.create_nothing_pickup(db.resource_database, "spr_ITEM_Nothing"),
-            self.players_config.player_index,
+    def _credits_spoiler(self) -> list:
+        pickups_to_show_in_credits = [
+            "Long Beam",
+            "Ice Beam",
+            "Wave Beam",
+            "Missile Launcher",
+            "Morph Ball",
+            "Bombs",
+            "Hi-Jump Boots",
+            "Varia Suit",
+        ]
+        for i in range(1, self.configuration.artifacts.required_artifacts + 1):
+            pickups_to_show_in_credits.append(f"Tourian Key {i}")
+
+        spoiler = []
+        spoiler_dict = credits_spoiler.generic_credits(
+            self.configuration.standard_pickup_configuration,
+            self.description.all_patches,
+            self.players_config,
+            PlanetsZebethHintNamer(self.description.all_patches, self.players_config),
+            "{}",
+            False,
+        )
+        for pickup_to_show_in_credits in pickups_to_show_in_credits:
+            if pickup_to_show_in_credits in spoiler_dict.keys():
+                if pickup_to_show_in_credits.startswith("Tourian Key "):
+                    if len([s_item for s_item in spoiler if s_item["header"] == "Tourian Key"]) == 0:
+                        spoiler.append(
+                            {
+                                "header": "Tourian Key",
+                                "description": [
+                                    spoiler_dict[f"Tourian Key {i}"]
+                                    for i in range(1, self.configuration.artifacts.required_artifacts + 1)
+                                ],
+                            }
+                        )
+                else:
+                    spoiler.append(
+                        {
+                            "header": pickup_to_show_in_credits,
+                            "description": spoiler_dict[pickup_to_show_in_credits].split("\n"),
+                        }
+                    )
```

Comment:
The main goal of `pickup_to_show_in_credits` was to order the pickups the way I want.
It would have been great if I could add `Big Missile Tank` too.

As for Tourian Keys, are the Metroid DNA considered generic in AM2R/MSR/Dread or is it a unique item per DNA?

Ex :
```
Metroid DNA

<location 1>
[...]
<location n>
```

or

```
Metroid DNA 1

<location>

[...]

Metroid DNA n

<location>
```

Reply:
Fine I'll use the spoiler dict only.

---

Just wanted to make sure that it's handled this way.